en/news/_posts/2016-06-20-ruby-2-4-0-preview1-released.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: news_post
+title: "Ruby 2.4.0-preview1 Released"
+author: "naruse"
+translator:
+date: 2016-06-20 14:00:00 +0000
+lang: en
+---
+
+We are pleased to announce the release of Ruby 2.4.0-preview1.
+
+Ruby 2.4.0-preview1 is the first preview of Ruby 2.4.0.
+This preview1 is released earlier than usual because it has so
+many new features and improvements.
+Feel free to [send feedback](https://bugs.ruby-lang.org/projects/ruby/wiki/HowToReport) since you can still change the features.
+
+## [Unify Fixnum and Bignum into Integer](https://bugs.ruby-lang.org/issues/12005)
+
+Though [ISO/IEC 30170:2012](http://www.iso.org/iso/iso_catalogue/catalogue_tc/catalogue_detail.htm?csnumber=59579) doesn't specify the detail of Integer,
+CRuby has two visible Integer classes: Fixnum and Bignum.
+Ruby 2.4 unifies them into Integer.
+
+## [String supports Unicode case mappings](https://bugs.ruby-lang.org/issues/10085)
+
+`String/Symbol#upcase/downcase/swapcase/capitalize(!)` now handle Unicode case mappings
+not only ASCII case mappings.
+
+## Performance improvements
+
+Ruby 2.4 also contains the following performance improvements including language changes:
+
+### [Array#max, Array#min](https://bugs.ruby-lang.org/issues/12172)
+
+In some condition, `[x, y].max` and `[x, y].min` are optimized
```

Comment:
`[x, y].max` and `[x, y].min` are optimized to not create a temporary array under certain conditions.